src/scenario.jl

```diff
@@ -368,29 +366,41 @@ function run_model(domain::Domain, param_set::NamedDimsArray, corals::DataFrame,
 
     p.mb .= corals.mb_rate  # background mortality
     @set! p.k = max_cover  # max coral cover
-    @set! p.comp = sim_params.comp  # competition rate
 
     # Proportionally adjust initial cover (handles inappropriate initial conditions)
     proportional_adjustment!(Y_cover[1, :, :], cover_tmp, max_cover)
+    
+    # Define constant table location for seed values and seeded coral types
+    taxa_to_seed = [2,3,5]
+    taxa_names = ["seed_CA","seed_TA","seed_SM"]
+
+    seeded_area = NamedDimsArray(zeros(length(taxa_to_seed)),taxa=taxa_names)
+    taxa_to_seed_ids = NamedDimsArray(corals.taxa_id .== taxa_to_seed',species=1:n_species,taxa=taxa_names)
 
-    # Define constant table location for seed values
-    tabular_enhanced::BitArray = corals.taxa_id .== 1
-    corymbose_enhanced::BitArray = corals.taxa_id .== 3
     target_class_id::BitArray = corals.class_id .== 2  # seed second smallest size class
-    seed_sc_TA::Int64 = first(findall(tabular_enhanced .& target_class_id))  # size class indices for TA and CA
-    seed_sc_CA::Int64 = first(findall(corymbose_enhanced .& target_class_id))
+    seed_sc = taxa_to_seed_ids .& target_class_id # size class indices for TA, CA and SM
 
-    # Extract colony areas for sites selected and convert to m^2
-    col_area_seed_TA = corals.colony_area_cm2[seed_sc_TA] / 10^4
-    col_area_seed_CA = corals.colony_area_cm2[seed_sc_CA] / 10^4
+    # Set up assisted adaptation values
+    a_adapt = zeros(n_species)
+    # Flag indicating whether to seed or not to seed
+    seed_corals = false
+    
+    # Extract colony areas for sites selected in m^2 and add adaptation values
+    for taxa in seeded_area.taxa
+        seeded_area[seeded_area.taxa.==taxa] .= colony_mean_area(corals.mean_colony_diameter_m[findall(seed_sc(taxa))])[1].*param_set(taxa)
```

Comment:
I'm not convinced `findall(seed_sc(taxa))])` is necessary here. Do you need it because `corals.mean_colony_diameter_m` is a NamedDimsArray or similar that has a quirk?

Actually, because you're looping over each element in `seeded_area`, this line doesn't seem necessary?

`seeded_area` is a three element vector, the order of which should align with the vectorized output of `colony_mean_area()` ?

Reply:
I'll see if I can vectorize and remove findall, it's mostly copied from the previous code